packages/native/types/base.d.ts

```diff
@@ -149,45 +152,58 @@ export interface CreateStyledComponent<
  */
 export interface CreateStyled<Theme extends {} = any> {
   <
-    C extends React.ComponentType<React.ComponentProps<C>>,
-    ForwardedProps extends keyof React.ComponentProps<
-      C
-    > = keyof React.ComponentProps<C>
+    Component extends ComponentType<ComponentPropsWithoutRef<Component>>,
+    ForwardedProps extends keyof ComponentPropsWithoutRef<
+      Component
+    > = keyof ComponentPropsWithoutRef<Component>
   >(
-    component: C,
-    options: FilteringStyledOptions<PropsOf<C>, ForwardedProps>
+    component: Component,
+    options: FilteringStyledOptions<
+      ComponentPropsWithoutRef<Component>,
+      ForwardedProps
+    >
   ): CreateStyledComponent<
-    Pick<PropsOf<C>, ForwardedProps> & { theme?: Theme },
+    Pick<ComponentPropsWithoutRef<Component>, ForwardedProps> & {
+      theme?: Theme
+    },
     {},
     { theme: Theme }
   >
 
-  <C extends React.ComponentType<React.ComponentProps<C>>>(
-    component: C,
-    options?: StyledOptions<PropsOf<C>>
-  ): CreateStyledComponent<PropsOf<C> & { theme?: Theme }, {}, { theme: Theme }>
+  <Component extends ComponentType<ComponentPropsWithoutRef<Component>>>(
+    component: Component,
+    options?: StyledOptions<ComponentPropsWithoutRef<Component>>
+  ): CreateStyledComponent<
+    ComponentPropsWithoutRef<Component> & { theme?: Theme },
+    {},
+    { theme: Theme }
+  >
 
   <
-    Tag extends keyof ReactNativeElements,
-    ForwardedProps extends keyof ReactNativeElements[Tag] = keyof ReactNativeElements[Tag]
+    ComponentName extends ReactNativeComponentNames,
+    ForwardedProps extends keyof ReactNativeComponentProps<
+      ComponentName
+    > = keyof ReactNativeComponentProps<ComponentName>
   >(
-    tag: Tag,
-    options: FilteringStyledOptions<ReactNativeElements[Tag], ForwardedProps>
+    component: ReactNativeComponents[ComponentName],
+    options: FilteringStyledOptions<
+      ReactNativeComponentProps<ComponentName>,
+      ForwardedProps
+    >
   ): CreateStyledComponent<
     { theme?: Theme },
-    Pick<ReactNativeElements[Tag], ForwardedProps>,
+    Pick<ReactNativeComponentProps<ComponentName>, ForwardedProps>,
     { theme: Theme }
   >
 
-  <Tag extends keyof ReactNativeElements>(
-    tag: Tag,
-    options?: StyledOptions<ReactNativeElements[Tag]>
+  <ComponentName extends ReactNativeComponentNames>(
+    component: ReactNativeComponents[ComponentName],
+    options?: StyledOptions<ReactNativeComponentProps<ComponentName>>
   ): CreateStyledComponent<
     { theme?: Theme },
-    ReactNativeElements[Tag],
+    ReactNativeComponentProps<ComponentName>,
     { theme: Theme }
   >
 }
 
-declare const styled: CreateStyled
-export default styled
+export const styled: CreateStyled
```

Comment:
as per the code in `base.js` the `styled` member is a named export, and there is no default export in that module.